src/com/cdd/bao/axioms/CellLineFix.java

```diff
@@ -0,0 +1,531 @@
+				
+				Set<String> terms = new HashSet<>();
+				for (int n = 0; n < annotList.length; n++)
+				{
+					String valueURI = annotList[n].optString("valueURI");
+					if (Util.isBlank(valueURI)) continue;
+					if (cellTerms.contains(valueURI)) terms.add(valueURI);
+				}
+				
+				int count = 0;
+				for (CellPair cell : pairs)
+				{
+					boolean has1 = terms.contains(cell.uri1), has2 = terms.contains(cell.uri2);
+					if ((!has1 && !has2) || (has1 && has2)) continue;
+					
+					if (count == 0) Util.writeln("[" + ze.getName() + "] {" + uniqueID + "} #annotations=" + annotList.length);
+					count++;
+					
+					String oldURI = null, oldLabel = null, newURI = null, newLabel = null;
+					if (has1 && !has2)
+					{
+						oldURI = cell.uri1;
+						oldLabel = cell.label1;
+						newURI = cell.uri2;
+						newLabel = cell.label2;
+					}					
+					else
+					{
+						oldURI = cell.uri2;
+						oldLabel = cell.label2;
+						newURI = cell.uri1;
+						newLabel = cell.label1;
+					}					
+
+					Util.writeln("    " + oldLabel + " <" + ModelSchema.collapsePrefix(oldURI) + "> " +
+							     "--> " + newLabel + " <" + ModelSchema.collapsePrefix(newURI) + ">");
+							     
+					lines.add(new String[]{aid, newURI, newLabel});
+				}
+				if (count > 0) toFix[0]++;
+			});
+		}
+		catch (UncheckedIOException e)
+		{
+			throw new IOException(e);
+		}
+		
+		Util.writeln("\nAssays needing changes: " + toFix[0]);
+		
+		File of = new File("term_additions.tsv");
+		Util.writeln("Writing these to: [" + of.getAbsolutePath() + "]");
+		try (Writer wtr = new FileWriter(of))
+		{
+			for (String[] bits : lines) wtr.write(String.join("\t", bits) + "\n");
+		}
+	}
+
+	private static String getProgramName()
+	{
+		StackTraceElement[] stack = Thread.currentThread().getStackTrace();
+		StackTraceElement main = stack[stack.length - 1];
+		String mainClass = main.getClassName();
+		return mainClass;
+	}
+
+	public static void main(String[] args)
+	{
+		org.apache.log4j.Logger.getRootLogger().setLevel(org.apache.log4j.Level.OFF);
+
+		Option curateOpt = Option.builder().longOpt("curate").desc("Path to zip file containing curated assays.").hasArg().build();
+		Option scoreOpt = Option.builder().longOpt("score").desc("Scoring sensitivity. Roughly, ignore matches with Levenshtein distances in excess of this score.").hasArg().build();
+		Option outfileOpt = Option.builder().longOpt("outfile").desc("Save semantic directives that make cell line corrections to the named file.").hasArg().required().build();
+
+		Options options = new Options();
+		options.addOption(curateOpt);
+		options.addOption(scoreOpt);
+		options.addOption(outfileOpt);
+
+		CommandLineParser parser = new DefaultParser();
+		CommandLine cmdLine = null;
+		try {cmdLine = parser.parse(options, args);}
+		catch (ParseException pe)
+		{
+			HelpFormatter formatter = new HelpFormatter();
+			formatter.printHelp(getProgramName(), options);
+			System.exit(1);
```

Comment:
Just use `return`.

Reply:
It makes sense to exit scripts that fail with error code 1. This is useful when the command line interface in an automated process (e.g. loading of assays during CI).

Reply:
Fair point, but none of the other command line entrypoints use this, and this particular use-case is quite explicitly designed to be hostile to automated scripting. (And I think it's been >20 years since I made use of an exit code... that pattern might be ready for garbage collection.)

Reply:
continuous integration depends on exit codes

Reply:
Do we have any plans to do that? All of our testing uses much newer technology (i.e. JUnit)

Reply:
In my view, having exit codes is always more valuable than otherwise.

Reply:
Or we slowly migrate our code to use exit codes when needed. I use the loading of assays in  continuous integration.

Reply:
I'm going to remove the `System.exit`s only because I value community rapport more than I do a rigorous adoption of exit standards.  But just to be clear:  any process with an ambiguous exit regime unanimously falls out of best practices.

Reply:
I only see several uses in the BAE code which I presume Peter put there, and I'm not sure what they do. There are dozens of exit places that just terminate without an exit. So basically don't use it until either: (1) we decide on a consistent policy, or (2) you have a real actual need for it.